### PR TITLE
Add nested resource reference for `RouteTable`

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
Adds resource references for `RouteTable.Routes` structs using the latest code-generator update

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by RedbackThomson** [comment]
/hold

Need to fix an error where the runtime will patch the concrete ID into the struct because the length and order of the routes as described by the API don't match the spec.

**Comment by RedbackThomson** [comment]
/unhold

I've pushed a fix that lets resource references work for the initial create. We need to fix https://github.com/aws-controllers-k8s/community/issues/1291 before we can go GA with this CRD, though. However, the solution for that issue will probably be in `runtime` and so there's no more work for here.

File: pkg/resource/route_table/hooks.go [hunk 1]
```diff
@@ -226,3 +230,29 @@ func (rm *resourceManager) addRoutesToStatus(
...
+	a *resource,
+	b *resource,
+) {
+	a.ko.Spec.Routes = removeLocalRoute(a.ko.Spec.Routes)
+	b.ko.Spec.Routes = removeLocalRoute(b.ko.Spec.Routes)
+}
+
+// removeLocalRoute will filter out any routes that have a gateway ID that
+// matches the local gateway.
+func removeLocalRoute(
```
Comment (brycahta) [inline pkg/resource/route_table/hooks.go:246]: It'd be helpful to include a detail about why this is needed in the func comment, ex: 

```
// removeLocalRoute will filter out any routes that have a gateway ID that
// matches the local gateway. Used to filter out the default local route, 
// which is always created by RouteTable and cannot be modified or deleted.
```

* [(AWS Docs) Every route table contains a local route...](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html#main-route-table)

**Comment by brycahta** [comment]
@RedbackThomson was the e2e test failure expected given the issue? Lmk if you need me to cleanup the test acct to re-run.

File: pkg/resource/route_table/hooks.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func (rm *resourceManager) syncRoutes(
 	toDelete := []*svcapitypes.CreateRouteInput{}
 
 	for _, desiredRoute := range desired.ko.Spec.Routes {
-		if *desiredRoute.GatewayID == LocalRouteGateway {
+		if (*desiredRoute).GatewayID != nil && *desiredRoute.GatewayID == LocalRouteGateway {
```
[the remark this answers is not in the document]
Comment (RedbackThomson) [reply pkg/resource/route_table/hooks.go:49]: Otherwise it attempts to dereference `(desiredRoute.GatewayID)`, which would panic.

**Review by brycahta — COMMENTED** [review]
/approve

**Review by brycahta — APPROVED** [review]
/lgtm

**Pull request merged**